src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -290,31 +299,14 @@
        (IllegalArgumentException.
          "'distinct_resources' query parameter requires accompanying parameters 'distinct_start_time' and 'distinct_end_time'")))))
 
-(defn warn-experimental
-  [entity warn-experimental]
-  (when (and warn-experimental
-             (contains? experimental-entities entity))
-    (log/warn (format
-                "The %s endpoint is experimental and may be altered or removed in the future."
-                (name entity)))))
-
 (defn query-route
-  "Returns a route for querying PuppetDB that supports the standard
-   paging parameters. Also accepts GETs and POSTs. Composes
-   `optional-handlers` with the middleware function that executes the
-   query."
-  [entity version param-spec & optional-handlers]
+  [version param-spec & optional-handlers]
   (let [handlers (or optional-handlers [identity])]
     (app
-      (extract-query param-spec)
-      (apply comp
-             (fn [{:keys [params globals puppetdb-query]}]
-               (warn-experimental entity (:warn-experimental globals))
```

Comment:
I pushed experimental warning on entities into the query engine, as it feels like a 'lower-down' problem.